scripts/inference/ft/mpt_ckpt_to_ft.py

```diff
@@ -0,0 +1,294 @@
+        val = val.T
+        split_vals = np.split(val, factor, axis=0)
+        for j in range(factor):
+            save_path = save_dir + '/model.' + key + '.%d.bin' % (i * factor +
+                                                                  j)
+            split_vals[j].tofile(save_path)
+        if config['no_bias']:
+            fake_weight_path = os.path.join(save_dir, 'model.' + key + '.bin')
+            write_empty_bias(key, fake_weight_path, val.shape[-1])
+
+    elif key.find('mlp.dense_h_to_4h.weight') != -1:
+        assert val.shape == (config['expansion_ratio'] * config['d_model'],
+                             config['d_model']), f'unexpected dim for {key}'
+        # nn.Linear weights are transposed
+        val = val.T
+
+        split_vals = np.split(val, factor, axis=-1)
+        for j in range(factor):
+            save_path = save_dir + '/model.' + key + '.%d.bin' % (i * factor +
+                                                                  j)
+            split_vals[j].tofile(save_path)
+            if config['no_bias']:
+                write_empty_bias(key, save_path, split_vals[j].shape[-1])
+
+    elif key.find('mlp.dense_h_to_4h.bias') != -1:
+        assert val.shape == (config['expansion_ratio'] *
+                             config['d_model'],), f'unexpected dim for {key}'
+        split_vals = np.split(val, factor, axis=-1)
+        for j in range(factor):
+            save_path = save_dir + '/model.' + key + '.%d.bin' % (i * factor +
+                                                                  j)
+            split_vals[j].tofile(save_path)
+
+    elif key.find('attention.query_key_value.bias') != -1:
+        assert val.shape == (3 *
+                             config['d_model'],), f'unexpected dim for {key}'
+
+        val = val.reshape(3, config['d_model'])
+
+        split_vals = np.split(val, factor, axis=-1)
+
+        for j in range(factor):
+            save_path = save_dir + '/model.' + key + '.%d.bin' % (i * factor +
+                                                                  j)
+            split_vals[j].tofile(save_path)
+
+    elif key.find('attention.query_key_value.weight') != -1:
+        assert val.shape == (3 * config['d_model'],
+                             config['d_model']), f'unexpected dim for {key}'
+        # nn.Linear weights are transposed
+        val = val.T
+
+        val = val.reshape(config['d_model'], 3, config['d_model'])
+        split_vals = np.split(val, factor, axis=-1)
+
+        for j in range(factor):
+            save_path = save_dir + '/model.' + key + '.%d.bin' % (i * factor +
+                                                                  j)
+            split_vals[j].tofile(save_path)
+            if config['no_bias']:
+                write_empty_bias(key, save_path, (3, split_vals[j].shape[-1]))
+
+    else:
+        print("[ERROR] cannot find key '{}'".format(key))
+
+
+def split_and_convert(args):
```

Comment:
We should design this API to be callable outside of a script. Can we make this main function named better, and have positional arguments? e.g. the desired external API would be:

```
def convert_to_faster_transformer(model_name_or_dir, ...)
```

Reply:
done.